src/ui/widget/image.rs

```diff
@@ -0,0 +1,137 @@
+//! Displays image to your users.
+
+use crate::graphics::{
+    self, Rectangle, Point,
+};
+use crate::ui::core:: {
+    Style, Node, Element, MouseCursor, Layout, Hasher, Widget,
+};
+
+use std::hash::Hash;
+
+/// A widget that displays an image.
+/// 
+/// It implements [`Widget`] when the associated [`core::Renderer`] implements
+/// the [`image::Renderer`] trait.
+///
+/// [`Widget`]: ../../core/trait.Widget.html
+/// [`core::Renderer`]: ../../core/trait.Renderer.html
+/// [`image::Renderer`]: trait.Renderer.html
+/// # Example
+///
+/// ```
+/// use coffee::graphics;
+/// use coffee::ui::Image;
+///
+/// let image_task = graphics::Image::load("resources/ui.png")
+/// 	.map(|image| Image::new(&image));
+/// ```
+#[derive(Debug)]
+pub struct Image {
+    image: graphics::Image,
+    source: Rectangle<u16>,
+    style: Style,
+}
+
+impl Image {
+    /// Creates a new [`Image`] with given image handle.
+    ///
+    /// [`Image`]: struct.Image.html
+    pub fn new(image: &graphics::Image) -> Self {
+        Image {
+            image: image.clone(),
+            source: Rectangle {
+                x: 0,
+                y: 0,
+                width: image.width(),
+                height: image.height(),
+            },
+            style: Style::default().fill_width().fill_height(),
+        }
+    }
+
+    /// Sets the portion of the [`Image`] that we want to draw.
+    /// 
+    /// [`Image`]: struct.Image.html
+    pub fn clip(mut self, source: Rectangle<u16>) -> Self {
```

Comment:
I'm not sure if I understood `clip` correctly, but if I did, the argument should be a `Rectangle`, not `Sprite` as it was written in #48

Reply:
Yes, you are right!